### PR TITLE
Add new positioning category for nav feedback

### PR DESCRIPTION
**Pull request by avi-c**
Fix for #2617

### CODE REVIEW

File: MapboxNavigation/FeedbackItem.swift [hunk 1]
```diff
@@ -81,6 +81,10 @@ public extension FeedbackType {
             return NSLocalizedString("ROUTE_QUALITY_ROAD_MISSING_FROM_MAP_FEEDBACK", bundle: .mapboxNavigation, value: "Road missing from map", comment: "Specifc route feedback that route contained road not shown on map.")
         case .roadClosure(.other):
             return NSLocalizedString("ROAD_CLOSURE_OTHER_FEEDBACK", bundle: .mapboxNavigation, value: "Other", comment: "Specific route feedback that a road closure type was encountered but not listed as a choice.")
+        case .positioning(.none):
+            return NSLocalizedString("POSITIONING", bundle: .mapboxNavigation, value: "Positioning", comment: "General category of route feedback where user position is incorrect.")
+        case .positioning(.userPosition):
+            return NSLocalizedString("POSITIONING", bundle: .mapboxNavigation, value: "User Position", comment: "Specific route feedback that user positioning is incorrect.")
```
Comment (1ec5) [inline MapboxNavigation/FeedbackItem.swift:87]: These two strings should have different keys.

Remember to run scripts/extract_localizable.sh to update the Localizable.strings file. Unfortunately, there won’t be much time to get translations of this string, but we’ll pull in any missing translations in v1.1.

File: MapboxNavigation/Resources/Base.lproj/Localizable.strings [hunk 2]
```diff
@@ -136,6 +136,12 @@
 /* Accessibility value of label indicating the absence of a rating */
 "NO_RATING" = "No rating set.";
 
+/* General category of route feedback where user position is incorrect. */
+"POSITIONING" = "Positioning";
+
+/* Specific route feedback that user positioning is incorrect. */
+"POSITIONING_USER" = "User Position";
```
Comment (1ec5) [inline MapboxNavigation/Resources/Base.lproj/Localizable.strings:143]: This string will be corrected in #2625.

**Review by 1ec5 — APPROVED** [review]

**Comment by avi-c** [comment]
@1ec5 - I merged everything into this single PR. Hopefully that's ok with you and once you review it we can merge this one.

File: MapboxCoreNavigation/EventDetails.swift [hunk 3]
```diff
@@ -352,7 +352,9 @@ enum EventDetailsError: Error {
 
 extension EventDetails {
     func asDictionary() throws -> [String: Any] {
-        let data = try JSONEncoder().encode(self)
+        let encoder = JSONEncoder()
+        encoder.keyEncodingStrategy = .convertToSnakeCase
+        let data = try encoder.encode(self)
```
Comment (1ec5) [inline MapboxCoreNavigation/EventDetails.swift:357]: Remember to remove this bit, since we don’t actually want the keys to be snake case after all.

**Comment by 1ec5** [comment]
Before merging, let’s find some way to verify that the JSON-encoded feedback event looks reasonably similar to the example Android feedback event that was shared.

**Comment by avi-c** [comment]
What I can say is that the events dictionary that I am now passing to the telem library does look similar to the Android JSON. It has the entry for the feedbackType and an array of subtypes. I haven't been able to see what is sent by the telem library or find an event in the server logs yet.

**Comment by avi-c** [comment]
The telem team has confirmed that one of the new events is indeed showing up on their stream, so I think we are ok to merge this.

File: MapboxCoreNavigation/EventDetails.swift (hunk 3 above)
Comment (1ec5) [reply MapboxCoreNavigation/EventDetails.swift:357]: #2628

**Comment by 1ec5** [comment]
Cherry-picked into the release-v1.0 branch as 18b89363fd8d36210372354383b034f1e65b6ddd.

**Pull request merged**